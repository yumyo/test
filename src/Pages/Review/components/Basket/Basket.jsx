  import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import omit from 'lodash/omit';
import BasketCard from '../BasketCard/BasketCard.jsx';
import './Basket.scss';

const mapStateToProps = (state) => Object({
    basket: state.basket,
});

export const BasketComponent = (props) => {
    const basket = omit(props, ['basket']);
    return (
        <div>
            <h1>Your Basket</h1>
            <p>You're currently looking at purchasing these tickets:</p>
            <div>
                { props.basket.map((item, index) => {
                    if (item && item.quantity) {
                        return (
                            <BasketCard
                                item={item}
                                key={`basket-item-${index}`}
                            />
                        );
                    }
                    return '';
                }) }
            </div>
        </div>
    );
};

BasketComponent.propTypes = {
    basket: PropTypes.array.isRequired,
};

export default connect(
    mapStateToProps,
)(BasketComponent);
