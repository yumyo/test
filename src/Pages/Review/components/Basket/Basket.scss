.basket {
  margin: 30px;
  padding: 30px;
  background: #f9eeb9;
}
