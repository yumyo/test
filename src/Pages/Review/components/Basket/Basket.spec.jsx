import React from 'react';
import { shallow } from 'enzyme';
import { BasketComponent } from './Basket.jsx';

describe('<Basket />', () => {
    let component;
    const props = {
        basket: [
            {
                id: '2',
                quantity: 1,
            }
        ],
    };

    beforeEach(() => {
        component = shallow(<BasketComponent {...props} />);
    });

    it('renders', () => {
        expect(component).toMatchSnapshot();
    });
});
