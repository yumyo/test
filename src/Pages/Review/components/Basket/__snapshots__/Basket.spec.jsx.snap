// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Basket /> renders 1`] = `
<div>
  <h1>
    Your Basket
  </h1>
  <p>
    You're currently looking at purchasing these tickets:
  </p>
  <div>
    <Connect(BasketCardComponent)
      item={
        Object {
          "id": "2",
          "quantity": 1,
        }
      }
      key="basket-item-0"
    />
  </div>
</div>
`;
